java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/consumer/BinaryConsumer.java

```diff
@@ -37,43 +35,39 @@
 
   private static final int BUFFER_SIZE = 1024;
 
-  private VarBinaryWriter writer;
+  private VarBinaryVector vector;
   private final int columnIndexInResultSet;
-  private BufferAllocator allocator;
 
-  private ArrowBuf reuse;
-  private byte[] reuseBytes;
+  private int currentIndex;
 
   /**
    * Instantiate a BinaryConsumer.
    */
   public BinaryConsumer(VarBinaryVector vector, int index) {
-    this.writer = new VarBinaryWriterImpl(vector);
```

Comment:
there seems to be a change in contract - earlier the writers were doing a setSafe() while handling the writes which would have expanded the buffers if needed.

buffer.SetBytes() would not i guess?

also if we want to error out no space - why not do the bounds checking once upfront and then we can use PlatformDependent to write the bytes

Reply:
Thanks, you are right, how about implement like this?

>if ((dataBuffer.writerIndex() + read) > dataBuffer.capacity()) {
          vector.reallocDataBuffer();
        }
        PlatformDependent.copyMemory(bytes, 0, dataBuffer.memoryAddress() + startIndex + totalBytes, read);

Reply:
fixed.